### PR TITLE
Add Rank Medal to APP

### PR DESCRIPTION
**Pull request by dkwl93**

### CODE REVIEW

File: app/components/ProfileCard.js [hunk 1]
```diff
@@ -43,10 +43,71 @@ class ProfileCard extends Component {
         }).catch(err => console.error('An error occured ', err));
     }
 
+    getRankIcon(rankTier, leaderboardRank) {
+        const rankIcons = {
```
Comment (sukwonchoi) [inline app/components/ProfileCard.js]: can port these into a rank_icon.js file? and export it there?

File: app/components/ProfileCard.js [hunk 2]
```diff
@@ -43,10 +43,71 @@ class ProfileCard extends Component {
...
+                {(star > 0) ? <Image style={styles.rankIcon} source={starIcons[star]}/> : null}
+            </View>
+        );
+    }
+
     render() {
         var info = this.props.info;
         var wl = this.props.wl;
         if(info.profile) {
+            const { rank_tier: rankTier, leaderboard_rank: leaderboardRank } = info;
```
Comment (sukwonchoi) [inline app/components/ProfileCard.js]: could destruct outside and include profile.

`const { profile, rank_tier: rankTier, leaderboard_rank: leaderboardRank } = this.props.info;`

File: app/components/ProfileCard.js [hunk 3]
```diff
@@ -43,10 +43,71 @@ class ProfileCard extends Component {
...
+        return (
+            <View style = {styles.rankContainer}>
+                <Image style={styles.rankIcon} source={rankIcons[rank]}/>
+                {(star > 0) ? <Image style={styles.rankIcon} source={starIcons[star]}/> : null}
+            </View>
+        );
+    }
+
     render() {
         var info = this.props.info;
```
Comment (sukwonchoi) [inline app/components/ProfileCard.js]: these can be `const` if you want to change it

**Review by sukwonchoi — CHANGES_REQUESTED** [review]
couple of suggestions

File: app/components/ProfileCard.js (hunk 1 above)
Comment (sukwonchoi) [reply app/components/ProfileCard.js]: and startIcons as well

File: app/components/ProfileCard.js [hunk 4]
```diff
@@ -43,10 +43,71 @@ class ProfileCard extends Component {
...
+        let rank, star;
+        if (rankTier > 9) {
+            if (leaderboardRank) {
+                if (leaderboardRank <= 10) {
+                    rank = '7c'; // Divine Top 10
+                } else if (leaderboardRank <= 100) {
+                    rank = '7b'; // Divine Top 100
+                }
+            }
+            if (rankTier === 76) {
```
Comment (Proyoyo) [inline app/components/ProfileCard.js]: @dkwl93 My brother has leaderboard rank of 2372, however, his rank tier is still 75 for some reason. I'm not sure if Valve changed how leaderboard rank works. For the time being, can you add a check to Divine elite if the rankTier is 75 and leaderboardRank isn't null?

screenshot_20180220-093426

File: app/components/ProfileCard.js [hunk 5]
```diff
@@ -43,10 +43,71 @@ class ProfileCard extends Component {
...
+                rank = rank || parseInt(intRankTier / 10, 10);
+            }
+        } else {
+            rank = '0';
+        }
+
+        return (
+            <View style = {styles.rankContainer}>
+                <Image style={styles.rankIcon} source={rankIcons[rank]}/>
+                {(star > 0) ? <Image style={styles.rankIcon} source={starIcons[star]}/> : null}
```
Comment (Proyoyo) [inline app/components/ProfileCard.js]: Can we show the leaderboard rank here as well for those that have leaderboardRank?

**Review by Proyoyo — CHANGES_REQUESTED** [review]
Added some feedback. Other than that, everything that @sukwonchoi mentioned above is valid.

File: app/components/ProfileCard.js (hunk 3 above)
Comment (dkwl93) [reply app/components/ProfileCard.js]: Was existing code lol

File: app/components/ProfileCard.js (hunk 2 above)
Comment (dkwl93) [reply app/components/ProfileCard.js]: done

File: app/components/ProfileCard.js (hunk 4 above)
Comment (dkwl93) [reply app/components/ProfileCard.js]: done

File: app/components/ProfileCard.js (hunk 5 above)
Comment (dkwl93) [reply app/components/ProfileCard.js]: done

File: app/components/ProfileCard.js (hunk 1 above)
Comment (dkwl93) [reply app/components/ProfileCard.js]: done

**Review by Proyoyo — APPROVED** [review]

**Pull request merged**